src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/volume/AzureVolumePlugin.java

```diff
@@ -83,9 +87,11 @@ public VolumeInstance getInstance(VolumeOrder volumeOrder, AzureUser azureUser)
         LOGGER.info(String.format(Messages.Info.GETTING_INSTANCE_S, volumeOrder.getInstanceId()));
         Azure azure = AzureClientCacheManager.getAzure(azureUser);
         String resourceName = AzureGeneralUtil.defineResourceName(volumeOrder.getInstanceId());
+        String resourceGroupName = AzureGeneralUtil
+                .selectResourceGroupName(azure, resourceName, this.defaultResourceGroupName);
```

Comment:
Is there an option to avoid to do another request to Azure cloud. 
If you jsut check the resourceName and the resourceNameGroup?

Reply:
Unfortunately, I not found another option.